llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -319,15 +291,43 @@ void SPIRVModuleAnalysis::collectFuncNames(MachineInstr &MI,
   }
 }
 
+using InstrSignature = SmallVector<size_t>;
+using InstrTraces = std::set<InstrSignature>;
```

Comment:
could use std::unordered_set here, since this code doesn't seem to care about ordering?

Reply:
It would be a problem to use with llvm, I see, for example: https://llvm.org/docs/ProgrammersManual.html#other-set-like-container-options